lib/awful/prompt.lua

```diff
@@ -461,17 +461,18 @@ function prompt.run(args, textbox, exe_callback, completion_callback,
                     for _,v2 in ipairs(v[1]) do
                         match = match and mod[v2]
                     end
-                    if match or #filtered_modifiers == 0 then
+                    if match then
```

Comment:
On purpose, it's an impossible condition. If `#filtered_modifiers == 0` then `#v[1] == 0` and we don't enter the previous loop, therefore `local match` remains true.